workflow/controller/operator.go

```diff
@@ -2651,3 +2679,48 @@ func (woc *wfOperationCtx) includeScriptOutput(nodeName, boundaryID string) (boo
 	}
 	return false, nil
 }
+
+func (woc *wfOperationCtx) fetchWorkflowTemplate() (*wfv1.WorkflowSpec, error) {
```

Comment:
```suggestion
func (woc *wfOperationCtx) fetchWorkflowSpec() (*wfv1.WorkflowSpec, error) {
```

This function actually returns a `WorkflowSpec`, not a `WorkflowTemplate`, so it should be named appropriately.